--- a/configure.ac
+++ b/configure.ac
@@ -55,6 +55,7 @@ FLTK_VERSION_PATCH=8
 FL_DSO_VERSION=${FLTK_VERSION_MAJOR}.${FLTK_VERSION_MINOR}
 FL_ABI_VERSION=${FLTK_VERSION_MAJOR}.${FLTK_VERSION_MINOR}.0
 FLTK_VERSION=${FLTK_VERSION_MAJOR}.${FLTK_VERSION_MINOR}.${FLTK_VERSION_PATCH}
+FLTK_VERSION_FULL=${FL_VERSION}
 
 AC_SUBST(FLTK_VERSION_MAJOR)
 AC_SUBST(FLTK_VERSION_MINOR)
@@ -62,6 +63,7 @@ AC_SUBST(FLTK_VERSION_PATCH)
 AC_SUBST(FL_DSO_VERSION)
 AC_SUBST(FL_ABI_VERSION)
 AC_SUBST(FLTK_VERSION)
+AC_SUBST(FLTK_VERSION_FULL)
 
 AC_SUBST(ARCHFLAGS)
 AC_SUBST(OPTIM)
@@ -1585,10 +1587,13 @@ dnl Set empty BINARY_DIR variable for fltk-config.in (CMake compatibility)
 BINARY_DIR=
 AC_SUBST(BINARY_DIR)
 
+INCLUDE_DIRS="${includedir}"
+AC_SUBST(INCLUDE_DIRS)
+
 dnl Write all of the files...
 AC_CONFIG_HEADER(config.h:configh.in)
 AC_CONFIG_HEADER(FL/abi-version.h:abi-version.in)
-AC_CONFIG_FILES([makeinclude fltk.list fltk-config fltk.spec FL/Makefile])
+AC_CONFIG_FILES([makeinclude fltk.list fltk-config fltk.spec FL/Makefile CMake/FLTKConfig.cmake])
 AC_OUTPUT
 
 dnl Make sure the fltk-config script is executable...
