--- a/configure.ac
+++ b/configure.ac
@@ -30,6 +30,7 @@ FLTK_VERSION_PATCH=$(echo AC_PACKAGE_VERSION | awk -F. '{print $3}')
 
 FL_DSO_VERSION="${FLTK_VERSION_MAJOR}.${FLTK_VERSION_MINOR}"
 FL_ABI_VERSION="${FLTK_VERSION_MAJOR}.${FLTK_VERSION_MINOR}.0"
+FLTK_VERSION_FULL="${FL_VERSION}"
 
 AC_SUBST(FLTK_VERSION)
 AC_SUBST(FLTK_VERSION_MAJOR)
@@ -37,6 +38,7 @@ AC_SUBST(FLTK_VERSION_MINOR)
 AC_SUBST(FLTK_VERSION_PATCH)
 AC_SUBST(FL_DSO_VERSION)
 AC_SUBST(FL_ABI_VERSION)
+AC_SUBST(FLTK_VERSION_FULL)
 
 
 dnl Required file in package...
@@ -1682,10 +1684,13 @@ dnl Set empty BINARY_DIR variable for fltk-config.in (CMake compatibility)
 BINARY_DIR=
 AC_SUBST([BINARY_DIR])
 
+INCLUDE_DIRS="${includedir}"
+AC_SUBST(INCLUDE_DIRS)
+
 dnl Write all of the files...
 AC_CONFIG_HEADERS([config.h:configh.in])
 AC_CONFIG_HEADERS([FL/fl_config.h:fl_config.in])
-AC_CONFIG_FILES([makeinclude fltk.list fltk-config fltk.spec FL/Makefile])
+AC_CONFIG_FILES([makeinclude fltk.list fltk-config fltk.spec FL/Makefile CMake/FLTKConfig.cmake])
 AC_OUTPUT
 
 dnl Make sure the fltk-config script is executable...
