diff --git a/setup.py b/setup.py
index 0e35ae7..6caf063 100644
--- a/setup.py
+++ b/setup.py
@@ -47,7 +47,7 @@ def install():
         python_requires=">=3.5.*",
         install_requires=install_requires,
         extras_require=extras_require,
-        packages=find_packages() + ["astroid.brain"],
+        packages=find_packages(exclude=['tests']) + ["astroid.brain"],
         setup_requires=pytest_runner,
         test_suite="test",
         tests_require=["pytest"],
